packages/gatsby/src/services/initialize.ts

```diff
@@ -367,9 +369,61 @@ export async function initialize({
 
   if (!oldPluginsHash || pluginsHash !== oldPluginsHash || cacheIsCorrupt) {
     try {
-      // Attempt to empty dir if remove fails,
-      // like when directory is mount point
-      await fs.remove(cacheDirectory).catch(() => fs.emptyDir(cacheDirectory))
+      const deleteGlobs = [
+        // By default delete all files & subdirectories
+        `${cacheDirectory}/**`,
+        `${cacheDirectory}/*/`,
+      ]
+
+      const sourceFileSystemVersion = require(`gatsby-source-filesystem/package.json`)
```

Comment:
this will throw if plugin is not installed ;)

Reply:
by this point we should have `flattenedPlugins` (or something like that) saturated in redux - we don't need to require - so we could find `gatsby-source-filesystem` and get the version from it if it exists

also it's not guaranteed we can resolve it from here depending how package manager installs it - we had a lot of bugs because of things like that with themes

Reply:
heh true — yeah I'll try to pull it off `flattenedPlugins`

Reply:
hmm that doesn't necessarily work actually as plugins can use `createRemoteFileNode` w/o gatsby-source-filesystem being in the gatsby-config.js — I'll try both and wrap the require w/ a try/catch